urwid/numedit.py

```diff
@@ -196,14 +253,18 @@ def __init__(
         self,
         caption="",
         default: str | int | Decimal | None = None,
-        preserveSignificance: bool = True,
-        decimalSeparator: str = '.',
+        preserveSignificance: bool | None = None,
+        decimalSeparator: str | None = None,
+        *,
+        preserve_significance: bool = True,
+        decimal_separator: str = ".",
+        allow_negative: bool = False,
```

Comment:
This appears to be a breaking change to code that used to use `preserveSignificance` and `decimalSeparator` as a keyword arg. The following changes should probably be made to preserve compatibility:

- Keep `preserveSignificance` defaulted to `True`
- Have `preserve_significance` default to `None`
- Assign `preserve_significance = preserveSignificance` if `preserve_significance == None`

And the same for `decimalSeparator`.

Otherwise, from the code, it looks like the value of `preserveSignificance` is ignored, which is the breaking part.

Either that or just leave it be and accept that the case doesn't fit the normal style.